doc/api/n-api.md

```diff
@@ -118,6 +118,96 @@ must make use exclusively of N-API by restricting itself to using
 and by checking, for all external libraries that it uses, that the external
 library makes ABI stability guarantees similar to N-API.
 
+## Building
+
+Unlike modules written in JavaScript, developing and deploying Node.js native
+addons using N-API requires an additional set of tools. Besides the basic
+tools required to develop for Node.js, the native addon developer requires a
+toolchain that can compile C and C++ code into a binary. In addition,
+depending upon how the native addon is deployed, the *user* of the native
+addon will also need to have a C/C++ toolchain installed.
+
+For Linux developers, the necessary C/C++ toolchain packages are readily
+available. For many developers, the [LLVM](https://llvm.org) compiler
+infrastructure is a good choice.
+
+For Mac developers, [Xcode](https://developer.apple.com/xcode/) offers all
+the required compiler tools. However, it is not necessary to install the
+entire Xcode IDE. The following command installs the necessary toolchain:
+
+```bash
+xcode-select --install
+```
+
+For Windows developers, [Visual Studio](https://visualstudio.microsoft.com)
+offers all the required compiler tools. However, it is not necessary to
+install the entire Visual Studio IDE. The following command installs the
+necessary toolchain:
+
+```bash
+npm install --global --production windows-build-tools
+```
+
+The sections below describe the additional tools available for developing
+and deploying Node.js native addons.
+
+### Build tools
+
+Both the tools listed here require that *users* of the native
+addon have a C/C++ toolchain installed in order to successfully install
+the native addon.
+
+#### node-gyp
+
+[node-gyp](https://github.com/nodejs/node-gyp) is a build system based on
+Google's [GYP](https://gyp.gsrc.io) tool and comes bundled with npm. GYP,
+and therefore node-gyp, requires that Python be installed.
+
+Historically, node-gyp has been the tool of choice for building native
+addons. It has widespread adoption and documentation. However, some
+developers have run into limitations in node-gyp.
+
+#### CMake.js
+
+[CMake.js](https://github.com/cmake-js/cmake-js) is an alternative build
```

Comment:
nit: there is a mixture of inline links such as these and bottom-of-document links in this document. We should use only bottom-of-document links – but I'm OK if we move that out of scope for this PR.

Reply:
I've move all links in the document to be end notes.